### DISCUSSION TITLE
Weapons not reported correctly in player_hurt

### DISCUSSION THREAD

**Discussion by dclayto1** [saul/demofile#288] [Questions] [answered]
I don't know if this is an issue on demofile's side or not, but it looks some weapons aren't reported correctly on the PlayerHurt

The 4 examples I have are:

- usp-s damage isn't reported as **usp_silencer**, but is as **hkp2000**
- m4a1-s damage isn't reported as **m4a1_silencer**, but is as **m4a1**
- mp5 damage isn't reported as **mp5sd**, but is as **mp7**
- r8 damage isn't reported as **revolver**, but is as **deagle**

The PlayerDeath event though _does_ have the right weapon. I was really just wondering if anyone else noticed this for the PlayerHurt event

**Comment by saul** [comment, marked as answer]
This is simply how the weapons are reported in the `player_hurt` event. If you want to get the actual weapon name, you need to get the attacking player entity, get their active weapon, then get the item name. Something like:

```ts
demoFile.gameEvents.on("player_hurt", e => {
  const victim = demoFile.entities.getByUserId(e.userid);
  const attacker = demoFile.entities.getByUserId(e.attacker);
  console.log(
    `${attacker?.name} hit ${victim?.name} [${attacker?.weapon?.itemName}] for ${e.dmg_health} HP`
  );
});
```

The game does this transformation for you already on `player_death`, but for some reason they didn't do it for `player_hurt`.

**Reply by dclayto1** [reply]
Ah okay, I wasn't sure where the disconnect was but figured it was how the game was reporting it. Thanks for the response and advice!!

**Reply by dclayto1** [reply]
In case anyone else runs into this same issue, to further accurately report the damage, you need to first distinguish whether the reported weapon is a grenade or not. If it's a grenade type, use the reported weapon, otherwise the damage would be attributed to what the attacker's current weapon is.
